tests/testTriggers.go

```diff
@@ -3698,6 +3702,96 @@ func TriggerAutoFsTrim(contexts *[]*scheduler.Context, recordChan *chan *EventRe
 
 }
 
+// TriggerVolumeUpdate enables to test volume update
+func TriggerVolumeUpdate(contexts *[]*scheduler.Context, recordChan *chan *EventRecord) {
+	defer ginkgo.GinkgoRecover()
+	event := &EventRecord{
+		Event: Event{
+			ID:   GenerateUUID(),
+			Type: UpdateVolume,
+		},
+		Start:   time.Now().Format(time.RFC1123),
+		Outcome: []error{},
+	}
+	defer func() {
+		event.End = time.Now().Format(time.RFC1123)
+		*recordChan <- event
+	}()
+	context("Validate update of the volumes", func() {
+
+		Step("Update Io priority on volumes ",
+			func() {
+				updateIOPriorityOnVolumes(contexts, event)
+				logrus.Info("Update IO priority call completed")
+			})
+	})
+}
+
+//This method is responsible for updating IO priority on Volumes if volume is labelled with
+//IO Priority. If lebelled with High and volume is updated to HIGH, next iteration it will reset
+//back to low.
+func updateIOPriorityOnVolumes(contexts *[]*scheduler.Context, event *EventRecord) {
+	for _, ctx := range *contexts {
+		var appVolumes []*volume.Volume
+		var err error
+		appVolumes, err = Inst().S.GetVolumes(ctx)
+		UpdateOutcome(event, err)
+		if len(appVolumes) == 0 {
+			UpdateOutcome(event, fmt.Errorf("found no volumes for app "))
+		}
+		for _, v := range appVolumes {
+			isPureVol, err := Inst().V.IsPureVolume(v)
```

Comment:
is this check required for iopriority?

Reply:
Yes. It is required as update IO priority does not support on FA or FADA volumes.